ui/src/containers/AppVersionsTableContainer.js

```diff
@@ -1,40 +1,69 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
-import { Checkbox } from '@patternfly/react-core';
-import { getApps, appDetailsSort } from '../actions/actions-ui';
+import { Checkbox, TextInput } from '@patternfly/react-core';
+import { getApps, appDetailsSort, updateDisabledApp, updateCustomMessage } from '../actions/actions-ui';
 import AppsTable from '../components/AppsTable';
 import './TableContainer.css';
 
 export class AppVersionsTableContainer extends React.Component {
+  componentWillMount () {}
+
+  handleCheckedChange = (_event, e) => {
+    const id = e.target.id;
+    const isDisabled = e.target.checked;
+    this.props.updateDisabledApp(id, isDisabled);
+  };
+
+  handleTextInputChange = (_event, e) => {
+    const id = e.target.id;
+    const value = e.target.value;
+    this.props.updateCustomMessage(id, value);
+  };
+
   onSort = (_event, index, direction) => {
     this.props.appDetailsSort(index, direction);
   };
 
+  createCheckbox = (id, checked) => {
+    return (
+      <React.Fragment>
+        <Checkbox
+          label=""
+          isChecked={checked}
+          onChange={this.handleCheckedChange}
+          aria-label="disable app checkbox"
+          id={id}
+        />
+      </React.Fragment>
+    );
+  };
+
+  createTextInput = (id, text) => {
+    return (
+      <React.Fragment>
+        <TextInput
+          id={id}
+          type="text"
+          placeholder="Add a custom message.."
+          value={text}
+          onChange={this.handleTextInputChange}
```

Comment:
@davidffrench I've left the above 2 lines in to be able to check the client model is updating correctly

Reply:
changed to onBlur now.